src/main/java/Action.java

```diff
@@ -0,0 +1,160 @@
+            super(action);
+        }
+
+        @Override
+        public void resolve(ArrayList<Task> taskList) {
+            if (super.action.equals("list")) {
+                System.out.println("Here are your tasks. You better start now:");
+                int length = taskList.size();
+                for (int i = 0; i < length; i++) {
+                    Task currentTask = taskList.get(i);
+                    int taskNumber = i + 1;
+                    String status = currentTask.status();
+                    System.out.println(taskNumber + "." + status);
+                }
+            } else {
+                System.out.println("Bye bro. See you soon.");
+            }
+        }
+    }
+
+    public static class NonTimedTaskAction extends Action { // todo
+        private final String taskDescription;
+
+        public NonTimedTaskAction(String taskDescription) {
+            super("todo");
+            this.taskDescription = taskDescription;
+        }
+
+        @Override
+        public void resolve(ArrayList<Task> taskList) {
+            Task newTask = new Task.ToDo(taskDescription);
+            Task.addTask(newTask, taskList);
+        }
+    }
+
+    public static class NumberedAction extends Action { // mark, unmark & delete
+        private final int taskNumber;
+
+        public NumberedAction(String action, int taskNumber) {
+            super(action);
+            this.taskNumber = taskNumber;
+        }
+
+        @Override
+        public void resolve(ArrayList<Task> taskList) throws IllegalArgumentException {
+            try {
+                if (super.action.equals("delete")) {
+                    Task.deleteTask(this.taskNumber, taskList);
+                } else if (super.action.equals("mark")) {
+                    Task.mark(this.taskNumber, taskList);
+                } else {
+                    Task.unmark(this.taskNumber, taskList);
+                }
+            } catch (IllegalArgumentException e) {
+                throw e;
+            }
+        }
+    }
+
+    public static class TimedTaskAction extends Action { // deadline & event
+        private final String taskDescription;
+        private final String timing;
+
+        public TimedTaskAction(String action, String taskDescription, String timing) {
+            super(action);
+            this.taskDescription = taskDescription;
+            this.timing = timing;
+        }
+
+        @Override
+        public void resolve(ArrayList<Task> taskList) {
+            Task newTask;
+
+            if (super.action.equals("deadline")) {
+                newTask =  new TimedTask.Deadline(this.taskDescription, this.timing);
+            } else {
+                newTask = new TimedTask.Event(this.taskDescription, this.timing);
+            }
+
+            Task.addTask(newTask, taskList);
+        }
+    }
+
+    public static Action actionParser(String input) throws UnknownActionException, EmptyTaskNameException,
```

Comment:
I like that you have this function to handle the commands given by the user. :) 
Maybe you can consider splitting the actions into separate functions and use abstraction to call them? Perhaps it would make the code for this function shorter and simpler to follow!